### PR TITLE
Make assets assignment sortable

### PR DESCRIPTION
**Pull request by danrot**
__Tasks:__

- [ ] test coverage
- [ ] gather feedback for my changes
- [ ] submit changes to the documentation
- [ ] added changelog line

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | yes
| Fixed tickets | none
| BC Breaks     | none
| Doc           | none

### CODE REVIEW

File: src/Sulu/Bundle/MediaBundle/Controller/MediaController.php [hunk 1]
```diff
@@ -96,21 +96,23 @@ public function cgetAction(Request $request)
...
-                'ids' => $ids,
-                'types' => $types,
-                'search' => $search
-            ), $limit, $offset);
+
+            if ($ids !== null) {
+                $ids = explode(',', $ids);
+                $media = $mediaManager->getByIds($ids, $this->getLocale($request));
+            } else {
+                $media = $mediaManager->get($this->getLocale($request), array(
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Controller/MediaController.php]: `s/media/medias`

File: src/Sulu/Bundle/MediaBundle/Controller/MediaController.php [hunk 2]
```diff
@@ -96,21 +96,23 @@ public function cgetAction(Request $request)
             $offset = ($request->get('page', 1) - 1) * $limit;
             $ids = $request->get('ids');
             $search = $request->get('search');
-            if ($ids !== null) {
-                $ids = explode(',', $ids);
-            }
             $types = $request->get('types');
             if ($types !== null) {
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Controller/MediaController.php]: Not part of this PR but it is never necessary to pass comment delimited values in a request. `you[can][encode]=arrays`

File: src/Sulu/Bundle/MediaBundle/Controller/MediaController.php (hunk 1 above)
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Controller/MediaController.php]: Hmm. actually media is plural isn't it.

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js [hunk 3]
```diff
@@ -674,6 +677,35 @@ define(['sulumedia/collection/collections', 'sulumedia/model/collection'], funct
             }
         },
 
+        initSortable = function() {
+            var $sortable = this.sandbox.dom.find('.sortable', this.$el),
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: Are you introducing a new `$convention` here?

Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: Or do we normally prefix dom-elements with `$` ?

File: src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php [hunk 4]
```diff
@@ -116,24 +118,49 @@ protected function setUpMedia()
...
+    protected function createMedia($name)
+    {
+        $media = new Media();
+        $media->setCreated(new DateTime());
+        $media->setChanged(new DateTime());
+        $media->setType($this->imageType);
+
+        // create file
+        $file = new File();
+        $file->setVersion(1);
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php]: Not part of this PR, but we could probably handle this with an event listener/subscriber.

File: src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php [hunk 5]
```diff
@@ -142,68 +169,58 @@ protected function setUpMedia()
...
-    protected function setUpCollection(&$collection)
+    protected function setUpCollection()
     {
+        $this->collection = new Collection();
         $style = array(
             'type' => 'circle',
             'color' => '#ffcc00'
         );
 
-        $collection->setStyle(json_encode($style));
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php]: fyi doctrine can handle this natively now: http://doctrine-dbal.readthedocs.org/en/latest/reference/types.html#json-array /cc @alexander-schranz

File: src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php [hunk 6]
```diff
@@ -256,6 +274,8 @@ public function testGetById()
      */
     public function testcGet()
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php]: `testCget`

Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php]: What does `cget` mean anyway?

File: src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php [hunk 7]
```diff
@@ -381,10 +408,33 @@ public function testcGetIds()
         $this->assertNotEmpty($response);
 
         $this->assertEquals(1, $response->total);
-        $this->assertEquals($this->media->getId(), $response->_embedded->media[0]->id);
+        $this->assertEquals($media->getId(), $response->_embedded->media[0]->id);
         $this->assertEquals('photo.jpeg', $response->_embedded->media[0]->name);
     }
 
+    public function testcGetMultipleIds()
```
Comment (dantleech) [inline src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php]: `testCget`

File: src/Sulu/Bundle/MediaBundle/Controller/MediaController.php (hunk 2 above)
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Controller/MediaController.php]: We have decided on a format of `/some/url?ids=3,5,6,7`, with your suggestions we would have to change that to `/some/url?id=3&id=5&id=6&id=7`, which is not as comfortable to type

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js (hunk 3 above)
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: jQuery objects are generally (or should) be prefixed with `$`.

File: src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php (hunk 6 above)
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php]: I guess it stands for collection get. Was not introduced by us, is a convention from the FOSRestBundle.

**Comment by danrot** [comment]
fixed

**Comment by dantleech** [comment]
I am having trouble testing this, it worked under Chromium, but I had problems under firefox. I do believe however, that this is a local problem for me. The failing test is not related to this, so feel free to merge.

**Comment by danrot** [comment]
@dantleech Can you merge it please?
